pyuvdata/uvdata/uvh5.py

```diff
@@ -567,6 +597,14 @@ def _read_header(
             # if lsts are in the background wait for them to return
             proc.join()
 
+        # Finally, backfill the apparent coords if they aren't in the original datafile
+        if self.phase_type == "phased" and (
+            (self.phase_center_app_ra is None)
+            or (self.phase_center_app_dec is None)
+            or (self.phase_center_app_pa is None)
+        ):
+            self._set_app_coords_helper()
```

Comment:
Are you using the presence of these to decide whether it used new vs old phasing? If so, does this prevent that?

Reply:
Ahh, good catch -- I've pushed this elsewhere and put in a `fix_phase` method in UVData that gets called by default when reading UV5H if these attributes are missing (although that can be overridden by setting `fix_old_proj=False`, in which case it'll just give you a warning). There is also a `fix_old_proj` parameter built into `read_uvfits` and `read_miriad`, but since it's not easy to identify off the bat if they were phased using the old method, the default for these methods is `fix_old_proj=False`. You can, after reading in any file, call `fix_phase` to undo the old correction.